### ISSUE TITLE
After a client disconnects, AP txpower drops - but only on 5Ghz

### ISSUE THREAD

**Issue by chrsmrtnx** [aircrack-ng/rtl8812au#286]
My EW-7822UAC (rtl8812au), running in AP mode, loses transmit power after the first client disconnects - but only on the 5Ghz band.

When hostapd starts the first time, the AP works well. With my phone, I can connect and browse, and the signal emitted is as strong as you would expect.

But if you disconnect from the AP (with your Android phone, say) then the signal strength from the AP drops to roughly half or a third of its original level, making it nearly useless (this is according to the wifi survey app on the phone). The problem only occurs on the 5GHz band. On 2.4GHz, you can connect and re-connect all you want and the AP signal strength stays the same. There are no obvious error messages in the system log.

A simple restart of hostapd makes no difference. To fix the problem, you have to rmmod/modprobe the 88XXau module and then restart hostapd. Alternatively, if you switch the AP to the 2.4Ghz band and then back to 5GHz (just by changing the hostapd settings and without reloading the module) the problem is also 'fixed'. That is, until your phone (or whatever) disconnects the first time, in which case the AP goes weak again, etc. etc.

"iw <wlan-iface-name> info" shows a steady (default) txpower of 18 dBm. Running "iw <wlan-iface-name> set txpower fixed 3000" boosts this to 30 dBm - or at least that's what reported back - but in practice this makes no difference.

So in summary, there appears to be some sort of driver bug where a disconnect messes up txpower - but only on 5Ghz. To fix the problem, the driver needs to be re-initialized, either by reloading the module, or by switching off of, then back to, the 5Ghz band.

This occurs with both the 5.2.20 and the 5.3.4 branches. I haven't tested others. I'm on kernel 4.19 (arm-based CuBox running Debian).

I'd appreciate your help. Thanks.

**Comment by kimocoder** [comment]
@chrsmrtnx 

You should move over to a newer branch, there are quite a few releases after v5.2.20(.2).
Hopefully that fixes your issue, if not.. feel free to re-open this issue :+1:  thanks!
